### ISSUE TITLE
Aggregate graph does not show other percentiles

### ISSUE THREAD

**Issue by DFlexy** [Cacti/cacti#5555]
I have a graph template that is applied to all graphs and aggregates, however when I create the normal graphs I can see the 95th and 99th ones but in the aggregates I cannot see this data as shown in the screenshots below

Normal graph
![image](https://github.com/Cacti/cacti/assets/50220928/2b7fbbac-dc59-428a-9612-d0b22b2c6105)

Agregate graph
![image](https://github.com/Cacti/cacti/assets/50220928/0e335c3b-4e17-457a-9420-149606b0639a)

Here model graph for aggregate
![image](https://github.com/Cacti/cacti/assets/50220928/25f90a71-6137-4604-8260-b03bbac641e5)

**Comment by TheWitness** [comment]
Yea, I'm grumbling about this.  The it can be included, but the logic is tricky.  It only works in a few cases, specifically when you are totaling all data sources.

**Comment by TheWitness** [comment]
So, update `lib/api_aggregate.php` from the `1.2.x` branch and then resave the Aggregate Template.  Let me know how it goes.  It should fix it.

**Comment by TheWitness** [comment]
I did one minor QA to also handle the hrules as well.

**Comment by DFlexy** [comment]
Hello, I made the adjustments to the lines and nothing, then I replaced the file and nothing.
I'm putting the debugs of the aggregated graphs with the problem and the debug of a simple graph working, remembering that they both use the same template

Version 1.2.25


![image](https://github.com/Cacti/cacti/assets/50220928/7f1aab23-ef55-4d07-a5bf-61dee11d493c)

Debug here
`
RRDtool Command:
/usr/bin/rrdtool graph - \
--imgformat=SVG \
--start='-86400' \
--end='-300' \
--pango-markup  \
--title='OTHERS' \
--vertical-label='bits por segundo' \
--slope-mode \
--base=1000 \
--height=150 \
--width=700 \
--alt-autoscale \
COMMENT:"From 19-11-2023 08\:43\:04 To 20-11-2023 08\:38\:04\c" \
COMMENT:"  \n" \
--color BACK#F3F3F3 \
--color CANVAS#FDFDFD \
--color SHADEA#CBCBCB \
--color SHADEB#999999 \
--color FONT#000000 \
--color AXIS#2C4D43 \
--color ARROW#2C4D43 \
--color FRAME#2C4D43 \
--border 1 \
--font TITLE:9: \
--font AXIS:8: \
--font LEGEND:9: \
--font UNIT:8: \
--font WATERMARK:8: \
--slope-mode \
--watermark 'Por ' \
DEF:a='/usr/share/cacti/rra/10_10_10_10_traffic_in_1148.rrd':'traffic_in':AVERAGE \
DEF:b='/usr/share/cacti/rra/10_10_10_10_traffic_in_1148.rrd':'traffic_out':AVERAGE \
DEF:c='/usr/share/cacti/rra/10_10_10_10_traffic_in_1238.rrd':'traffic_in':AVERAGE \
DEF:d='/usr/share/cacti/rra/10_10_10_10_traffic_in_1238.rrd':'traffic_out':AVERAGE \
CDEF:cdefa='a,0,*' \
CDEF:cdefb='b,0,*' \
CDEF:cdefc='c,0,*' \
CDEF:cdefd='d,0,*' \
CDEF:cdefe='TIME,1700480284,GT,a,a,UN,0,a,IF,IF,TIME,1700480284,GT,c,c,UN,0,c,IF,IF,+,8,*' \
CDEF:cdefi='TIME,1700480284,GT,b,b,UN,0,b,IF,IF,TIME,1700480284,GT,d,d,UN,0,d,IF,IF,+,8,*' \
AREA:cdefa#306EFF7F:  \
AREA:cdefb#E8A3177F:  \
AREA:cdefc#306EFF7F:  \
AREA:cdefd#E8A3177F:  \
AREA:cdefe#306EFF7F:'Total Inbound '  \
GPRINT:cdefe:LAST:'Current\:%8.2lf %s'  \
GPRINT:cdefe:AVERAGE:'Average\:%8.2lf %s'  \
GPRINT:cdefe:MAX:'Maximum\:%8.2lf %s\n'  \
AREA:cdefi#E8A3177F:'Total Outbound'  \
GPRINT:cdefi:LAST:'Current\:%8.2lf %s'  \
GPRINT:cdefi:AVERAGE:'Average\:%8.2lf %s'  \
GPRINT:cdefi:MAX:'Maximum\:%8.2lf %s\n'  \
COMMENT:' \n'  \
HRULE:62026371223.2#800080FF:'99th Percentile In  ->'
RRDtool Command lengths = 1760 characters.
RRDtool Says:
SVG/XML Output OK
`

**Comment by DFlexy** [comment]
Simple graph here OK

![image](https://github.com/Cacti/cacti/assets/50220928/e27ca32d-fc11-454d-8059-9793725b7b72)

Debug here
`
RRDtool Command:
/usr/bin/rrdtool graph - \
--imgformat=SVG \
--start='-86400' \
--end='-300' \
--pango-markup  \
--title='10.10.10.10 / ae4.0 / 10000 / OTHER - 10G' \
--vertical-label='bits por segundo' \
--slope-mode \
--base=1000 \
--height=150 \
--width=700 \
--alt-autoscale \
COMMENT:"From 19-11-2023 08\:44\:44 To 20-11-2023 08\:39\:44\c" \
COMMENT:"  \n" \
--color BACK#F3F3F3 \
--color CANVAS#FDFDFD \
--color SHADEA#CBCBCB \
--color SHADEB#999999 \
--color FONT#000000 \
--color AXIS#2C4D43 \
--color ARROW#2C4D43 \
--color FRAME#2C4D43 \
--border 1 \
--font TITLE:9: \
--font AXIS:8: \
--font LEGEND:9: \
--font UNIT:8: \
--font WATERMARK:8: \
--slope-mode \
--watermark 'Por ' \
DEF:a='/usr/share/cacti/rra/10_10_10_10_traffic_in_1355.rrd':'traffic_in':MAX \
DEF:b='/usr/share/cacti/rra/10_10_10_10_traffic_in_1355.rrd':'traffic_in':AVERAGE \
DEF:c='/usr/share/cacti/rra/10_10_10_10_traffic_in_1355.rrd':'traffic_out':MAX \
DEF:d='/usr/share/cacti/rra/10_10_10_10_traffic_in_1355.rrd':'traffic_out':AVERAGE \
CDEF:cdefa='a,8,*' \
CDEF:cdefb='b,8,*' \
CDEF:cdeff='c,8,*' \
CDEF:cdefg='d,8,*' \
LINE1:cdefa#306EFFFF: \
AREA:cdefb#306EFF7F:'Inbound '  \
GPRINT:cdefb:LAST:'Current\:%8.2lf %s'  \
GPRINT:cdefb:AVERAGE:'Average\:%8.2lf %s'  \
GPRINT:cdefb:MAX:'Maximum\:%8.2lf %s\n'  \
LINE1:cdeff#E8A317FF: \
AREA:cdefg#E8A3177F:'Outbound'  \
GPRINT:cdefg:LAST:'Current\:%8.2lf %s'  \
GPRINT:cdefg:AVERAGE:'Average\:%8.2lf %s'  \
GPRINT:cdefg:MAX:'Maximum\:%8.2lf %s\n'  \
COMMENT:' \n'  \
HRULE:269789501.16#000000FF:'95th Percentile In  ->' \
COMMENT:'(269.79 mbit)'  \
HRULE:463411527.58#800080FF:'99th Percentile In  ->' \
COMMENT:'(463.41 mbit)\n'  \
HRULE:269789501.16#E41B17FF:'95th Percentile Out ->' \
COMMENT:'(48.44 mbit)'  \
HRULE:53565025.77#C04000FF:'99th Percentile Out ->' \
COMMENT:'(53.57 mbit)\n' 
RRDtool Command lengths = 1793 characters.
RRDtool Says:
SVG/XML Output OK
`

**Comment by TheWitness** [comment]
So, are you saying it's working on some aggregates and not others?

**Comment by DFlexy** [comment]
> So, are you saying it's working on some aggregates and not others?

not only in simple graphs, aggregates do not work

**Comment by TheWitness** [comment]
Now I'm confused.  Do you have any aggregates that after you resave either the aggregate or aggregate template that are working?  Also, what happens when you create a new aggregate graph?

**Comment by DFlexy** [comment]
> Now I'm confused. Do you have any aggregates that after you resave either the aggregate or aggregate template that are working? Also, what happens when you create a new aggregate graph?

I have an aggregation template that uses the same template as normal graphs, but normal graphs work and aggregates do not use the same template

**Comment by DFlexy** [comment]
Here normal template graph
![image](https://github.com/Cacti/cacti/assets/50220928/d97dfd08-45f0-4cbb-9972-de715ae0c429)

template agregatte after change.
Bugged

![image](https://github.com/Cacti/cacti/assets/50220928/d37255a1-32e7-4f0b-8203-38d20b5d904b)

**Comment by DFlexy** [comment]
Template agreggate after change bugged

![image](https://github.com/Cacti/cacti/assets/50220928/1ebdc143-2106-4bd8-b4f4-93953a17ba00)

template normal graph ok
![image](https://github.com/Cacti/cacti/assets/50220928/10ad12d6-8ae6-41b2-9ce1-7cdcc0b124af)

**Comment by DFlexy** [comment]
To create the aggregate I use the same graph template which is

![image](https://github.com/Cacti/cacti/assets/50220928/ce306edc-c627-414a-9f8b-fb11a15c53a4)

**Comment by DFlexy** [comment]
ROLLBACK api_aggregate.php and see agregate template for info

![image](https://github.com/Cacti/cacti/assets/50220928/337008a8-1461-4b72-bba0-41effc8e65f7)

**Comment by TheWitness** [comment]
Yea, that modified aggregate was a yikes moment.  I'll dig in later today.

**Comment by DFlexy** [comment]
> Yea, that modified aggregate was a yikes moment. I'll dig in later today.

I am very grateful for the help, this has been bothering me a lot because I need the percentiles of the aggregates.
Thank you for the support

**Comment by TheWitness** [comment]
Can you please screen capture your Aggregate Template.  I need the "Aggregate Template Settings" section.  Thanks.  I'm still finding the Aggregates a little buggy, though I was not able to reproduce your issue.

**Comment by DFlexy** [comment]
> Can you please screen capture your Aggregate Template. I need the "Aggregate Template Settings" section. Thanks. I'm still finding the Aggregates a little buggy, though I was not able to reproduce your issue.

Here

![image](https://github.com/Cacti/cacti/assets/50220928/3f2cf1d7-e0b4-4de4-9723-34cbcae53e72)

**Comment by DFlexy** [comment]
Here Graph simple too if need
![image](https://github.com/Cacti/cacti/assets/50220928/d5fa15aa-15e9-4340-b3d5-6ce51b5e5287)

**Comment by TheWitness** [comment]
Okay, tested on my side.  I have a few updates to commit yet.

![image](https://github.com/Cacti/cacti/assets/1439914/6c48600a-03fc-4bca-962d-82a865869fbf)

**Comment by DFlexy** [comment]
> Okay, tested on my side. I have a few updates to commit yet.
> 
> ![image](https://user-images.githubusercontent.com/1439914/284286267-6c48600a-03fc-4bca-962d-82a865869fbf.png)

You can place the IN and OUT separately in the aggregates like in the simple graphs because adding the in and out is wrong

**Comment by TheWitness** [comment]
Yea, you 'should' find that fixed in this latest update.  Make sure you are pulling from the 1.2.x branch.  3 files this time.

**Comment by DFlexy** [comment]
I replaced the 3 files and now no percentile appears anymore

![image](https://github.com/Cacti/cacti/assets/50220928/3872cf5c-6fad-42d7-ab78-afc339c7d008)

**Comment by DFlexy** [comment]
W8 create one new agregate i thing work moment

**Comment by DFlexy** [comment]
It didn't work for the old units
But I created a new one and it worked, but only in the sense of IN

![image](https://github.com/Cacti/cacti/assets/50220928/4cd3266f-65bf-4416-ae5b-a5347bca4606)

**Comment by TheWitness** [comment]
I think I may know what is happening.  The confluence of items is coming from the member graphs, if the member graphs are not the Nth percentile graphs, then the aggregate will not be constructed properly.  A bit of an oversight.  I'll get to this later tonight if all goes well.

**Comment by TheWitness** [comment]
Okay, I just updated `lib/api_aggregate.php`.  Remember to save your Aggregate Template again.

**Comment by DFlexy** [comment]
> Okay, I just updated `lib/api_aggregate.php`. Remember to save your Aggregate Template again.

I changed the file and saved the aggregate template but it didn't work.
I created another aggregate template to test and when I created an aggregate chart using the template it came in the same way only the IN

Here new graph with new model template
![image](https://github.com/Cacti/cacti/assets/50220928/a7318934-f9ea-46b8-93f4-6b7333f535e6)

![image](https://github.com/Cacti/cacti/assets/50220928/50efbf9c-45fb-4318-bf16-c5e548785657)

**Comment by TheWitness** [comment]
Okay, I have it now...  Committing in a few minutes.  I did not see this before.

**Comment by TheWitness** [comment]
Okay, it should work now.  Now that this is fixed, let's talk about what else is essentially broken...  Give me more details, but do it in another ticket Okay?

**Comment by TheWitness** [comment]
Whoops, forgot to test before committing.

**Comment by TheWitness** [comment]
This looks way better now...

![image](https://github.com/Cacti/cacti/assets/1439914/2de60f24-06c8-4c8f-b279-9a500dfd7cee)

**Comment by DFlexy** [comment]
I can't access the aggregate option now in the template menu
![image](https://github.com/Cacti/cacti/assets/50220928/1f45d44b-776e-4683-a1c3-418e9282661a)

**Comment by TheWitness** [comment]
Grap a fresh copy.  I have committed a typo.

**Comment by DFlexy** [comment]
The options worked but did not bring the values

![image](https://github.com/Cacti/cacti/assets/50220928/69d893f0-6267-4073-90e0-fc6d61f4bccd)

**Comment by TheWitness** [comment]
Inch by inch ;)

**Comment by TheWitness** [comment]
Cross you fingers...

**Comment by DFlexy** [comment]
equal to the values of IN
it's almost lol

![image](https://github.com/Cacti/cacti/assets/50220928/798d69e0-ac73-4ee7-b21f-50a0bd6b8728)

**Comment by TheWitness** [comment]
Ugh.  I'm seeing a few other issues as well.  Still digging in.

**Comment by TheWitness** [comment]
Squeaking almost into new feature territory now.  Assessing it.  Have to check the underlying library.

**Comment by TheWitness** [comment]
Grab one more updated.  Make sure you save your Aggregate Template again.

**Comment by TheWitness** [comment]
I'm going to have to put this code in a vault when done.  Lot's of moving parts.

**Comment by DFlexy** [comment]
Nothing now

![image](https://github.com/Cacti/cacti/assets/50220928/03d39ca5-7c2c-43d2-aa31-61dae590625d)

**Comment by TheWitness** [comment]
Okay, one more update.  This updates: `lib/graph_variables.php` and `lib/rrd.php`

**Comment by TheWitness** [comment]
Sorry, there were not logged to this ticket.

**Comment by DFlexy** [comment]
Where are the files for me to get?

**Comment by TheWitness** [comment]
1.2.x branch.

**Comment by TheWitness** [comment]
https://github.com/Cacti/cacti/commit/914f187fbba89965cc7ab314dbe6ad0020eaa3ed

**Comment by TheWitness** [comment]
Crap missed one change.  You have to re-save again.  Sorry, long day.

**Comment by DFlexy** [comment]
I changed the 3 files and still no data
and saved the template again

My time on the company's VPN is running out, I'll only be able to test it tomorrow

**Comment by TheWitness** [comment]
Okay, yea, it's broken for me too.  Moving too fast.  Jump back in tomorrow.  We still have a few more days till release.

**Comment by DFlexy** [comment]
> Okay, yea, it's broken for me too. Moving too fast. Jump back in tomorrow. We still have a few more days till release.

Thank you for the great support have a great day tomorrow we are back

**Comment by TheWitness** [comment]
Okay, you have two more files to download, and then you must again save the Template.

**Comment by DFlexy** [comment]
Good morning, let's get to the facts.

I deleted the aggregate model and created a new one.
And I arrived at the following result.

When I added just 2 interfaces it worked ;)
When I added just 4 interfaces, the percentiles disappeared.
When I added more than 75 interfaces it worked.

And another point is that I am having several ERROR logs as below

Another point when passing Giga could make the information more summarized instead of 1590571.06 mbit

TEST WITH 2 INTERFACES
![image](https://github.com/Cacti/cacti/assets/50220928/d05d25d3-e46e-4f20-8c7e-87fc8f429146)

TEST WITH 4 INTERFACES
![image](https://github.com/Cacti/cacti/assets/50220928/15eb00be-fa3f-4e4d-8f8d-4ac9a00860f6)

TEST WITH 75 INTERFACES (SLOW)
![image](https://github.com/Cacti/cacti/assets/50220928/5ea2cc5e-5f0e-4aab-90e2-ab688f1faad4)

ERROS LOG
![image](https://github.com/Cacti/cacti/assets/50220928/9d3d4e15-aafd-4ca1-9f88-7ebc213bae07)

**Comment by TheWitness** [comment]
That's trace code.  You may be one version off.  I'll check to ensure it was committed.  You too. You should not be seeing that debug code any longer.

**Comment by DFlexy** [comment]
I went to the aggregate template and clicked save and now the percentiles are gone again.

**Comment by DFlexy** [comment]
When I click on save, the aggregate of 75 interfaces works normally, only the other 2 and 4 interfaces no longer show the percentiles.

**Comment by TheWitness** [comment]
Funny and sad at the same time.  If I were not laughing, I'd be crying otherwise.

Relative to speed, moving aggregate to a using VDEFs would make it way faster.  Not in 1.2.x though.  It's a big project.

**Comment by TheWitness** [comment]
I'm still horizontal.  Once I get to a computer, I'll double check the code for the trace logic.

**Comment by DFlexy** [comment]
same template

![image](https://github.com/Cacti/cacti/assets/50220928/cef96bd0-c859-4281-afac-7a2fa21255d5)

**Comment by TheWitness** [comment]
I did miss one important update.  Please apply it.  You should probably take all three files from 1.2.x: `lib/rrd.php`, `lib/graph_variables.php`, and `lib/api_aggregate.php`.  Your Cacti log should be clean afterwards.  Keep me posted.

**Comment by DFlexy** [comment]
Logs clean ;)

**Comment by DFlexy** [comment]
Update with 20 interfaces

only 40 works
2 and 4 and 20 doesn't work

![image](https://github.com/Cacti/cacti/assets/50220928/3c33a38f-1871-45de-b683-267c1424d111)

**Comment by TheWitness** [comment]
Okay, once my day cools, I'll look for more clues.

**Comment by TheWitness** [comment]
Okay, so on those failed cases, can you describe the Graphs that they are coming from?  Are they the same Graph Template?  In other words, do they all have the 4 percentiles or not.  This is very important.

**Comment by TheWitness** [comment]
I'll be off tomorrow and we can do another round of interactive testing if you want.  Even a zoom.

**Comment by DFlexy** [comment]
Good morning.
The graphics are the same.
The following is happening, when I create the graph, always using the template, the percentiles appear, however, after I click on save the graph or save the template, the graphs are losing the percentiles, except for the one with 40 interfaces.

**Comment by TheWitness** [comment]
Are you okay with a Zoom?

**Comment by TheWitness** [comment]
If so, send your details to thewitness@cacti.net and I'll send an invite this morning.

**Comment by DFlexy** [comment]
I don't speak English and my company's VPN only uses Teams

**Comment by TheWitness** [comment]
We don't need audio.  Just send a compatible link out and I can join that way.  We can use Chat.  How are your vim skills?

**Comment by TheWitness** [comment]
@DFlexy, Update `lib/api_aggregate.php`.  If it does not fix, consider sanitizing anything confidential and send me a dump of the database via Email.  That way it's easier for me to reproduce.

**Comment by DFlexy** [comment]
Bug here again

![image](https://github.com/Cacti/cacti/assets/50220928/3d143ede-0da6-490c-9227-94959a596d1b)
![image](https://github.com/Cacti/cacti/assets/50220928/38d48250-35dd-4fc9-b74e-93e952e10b83)

**Comment by TheWitness** [comment]
What line in what file?

**Comment by TheWitness** [comment]
Run the audit_database.php script and correct the issues.

**Comment by TheWitness** [comment]
Please send me your database dump if you want this fixed before the 1.2.26 release.  You know where.

**Comment by DFlexy** [comment]
I made the exchange and ran the following tests.

When I create the aggregate it works normally
However, when I save the template, the percentiles of the aggregate created only.
But when you go to the created aggregate and remove it from the template model by clicking on Propagation Enabled and unchecking the option, the percentiles will return and it will be left out of the model


After creating the aggregate using the model
![image](https://github.com/Cacti/cacti/assets/50220928/623cb349-8489-49a2-89c9-72879142f61e)

![image](https://github.com/Cacti/cacti/assets/50220928/98817b45-5269-4665-ac8d-d43a76b5f571)

After save model agregate
![image](https://github.com/Cacti/cacti/assets/50220928/4b80a583-1891-444a-acb4-abb6a6e3cd1b)

After remove option Propagation Enabled
![image](https://github.com/Cacti/cacti/assets/50220928/a4dc8485-9a4a-4dd4-8cad-543d9b891ffd)

![image](https://github.com/Cacti/cacti/assets/50220928/0a6ecd50-fbde-4ad3-a461-983ddd58cacc)

![image](https://github.com/Cacti/cacti/assets/50220928/7db37f47-0263-434a-b917-80e5d11f416c)

**Comment by TheWitness** [comment]
Well, that's a very interesting development.

**Comment by TheWitness** [comment]
Is it possible to send the dump?

**Comment by DFlexy** [comment]
> Is it possible to send the dump?

I tried to remove the database dump but I can't remove the confidential information. Do I need a specific table?

**Comment by TheWitness** [comment]
One thing you can do is edit the sql file and then search for confidential information.  If you edit in VIM, you can replace as follows:

```sh
vim mydump.sql
# Assume you are searching to replace "ABC" with something else
:%s/ABC/randomsomething/g
# That above command replaces everything in the file that matches "ABC" with the string "randomsomething"
```
Just keep doing that until all the 'confidential' stuff is gone.

**Comment by TheWitness** [comment]
Another approach is just use sed

```sh
sed -i 's/ABC/somethingrandom/g' mydump.sql
```

**Comment by TheWitness** [comment]
Things confidential can be considered: ip addresses, snmp communities, site names, hostnames, and even interface names.  I only have interest in the situation.  I don't care about the rest.

**Comment by TheWitness** [comment]
The other route is this.  When you view an aggregate graph, you find in the address bar and id like in the case below, it's `1032`.  That is the local graph id of the Aggregate graph.  So, when the graphs are broken, run this SQL command below:

```sql
select task_item_id, value, text_format from graph_templates_item where local_graph_id=1032;
```

Then, fix them, and run the command again as above.  Send the output from each to the ticket.

**Comment by TheWitness** [comment]
Here is my example:

```sh
MariaDB [cacti]> select task_item_id, value, text_format from graph_templates_item where local_graph_id=1032;                                        +--------------+--------------------------------------+----------------------------------------------------+
| task_item_id | value                                | text_format                                        |
+--------------+--------------------------------------+----------------------------------------------------+
|         2268 |                                      |                                                    |
|         2270 |                                      |                                                    |
|         2264 |                                      |                                                    |
|         2272 |                                      |                                                    |
|         2274 |                                      |                                                    |
|         2276 |                                      |                                                    |
|         2278 |                                      |                                                    |
|         2266 |                                      |                                                    |
|         2280 |                                      |                                                    |
|         2282 |                                      |                                                    |
|         2269 |                                      |                                                    |
|         2271 |                                      |                                                    |
|         2265 |                                      |                                                    |
|         2273 |                                      |                                                    |
|         2275 |                                      |                                                    |
|         2277 |                                      |                                                    |
|         2279 |                                      |                                                    |
|         2267 |                                      |                                                    |
|         2281 |                                      |                                                    |
|         2283 |                                      |                                                    |
|          123 |                                      | Total Inbound                                      |
|          123 |                                      | Current:                                           |
|          123 |                                      | Average:                                           |
|          123 |                                      | Maximum:                                           |
|          124 |                                      | Total Outbound                                     |
|          124 |                                      | Current:                                           |
|          124 |                                      | Average:                                           |
|          124 |                                      | Maximum:                                           |
|            0 |                                      |                                                    |
|          123 | |95:bits:0:aggregate_current_peak:2| | 95th Percentile                                    |
|          123 |                                      | (|95:bits:6:aggregate_current_peak:2| mbit in+out) |
|          123 | |98:bits:0:aggregate_current_peak:2| | 98th Percentile                                    |
|          123 |                                      | (|98:bits:6:aggregate_current_peak:2| mbit in+out) |
+--------------+--------------------------------------+----------------------------------------------------+
33 rows in set (0.001 sec)
```

**Comment by DFlexy** [comment]
This is the result of the aggregated graph of 75 interfaces that does not cause problems and always works

```
MariaDB [cactidb]> select task_item_id, value, text_format from graph_templates_item where local_graph_id=1477;
+--------------+---------------------------------+----------------------------------------+
| task_item_id | value                           | text_format                            |
+--------------+---------------------------------+----------------------------------------+
|          130 |                                 |                                        |
|          131 |                                 |                                        |
|          132 |                                 |                                        |
|          133 |                                 |                                        |
|          134 |                                 |                                        |
|          135 |                                 |                                        |
|          182 |                                 |                                        |
|          183 |                                 |                                        |
|          212 |                                 |                                        |
|          213 |                                 |                                        |
|          214 |                                 |                                        |
|          215 |                                 |                                        |
|          216 |                                 |                                        |
|          217 |                                 |                                        |
|          228 |                                 |                                        |
|          229 |                                 |                                        |
|          230 |                                 |                                        |
|          231 |                                 |                                        |
|          232 |                                 |                                        |
|          233 |                                 |                                        |
|          306 |                                 |                                        |
|          307 |                                 |                                        |
|          312 |                                 |                                        |
|          313 |                                 |                                        |
|          322 |                                 |                                        |
|          323 |                                 |                                        |
|          324 |                                 |                                        |
|          325 |                                 |                                        |
|          326 |                                 |                                        |
|          327 |                                 |                                        |
|          328 |                                 |                                        |
|          329 |                                 |                                        |
|          330 |                                 |                                        |
|          331 |                                 |                                        |
|          332 |                                 |                                        |
|          333 |                                 |                                        |
|          382 |                                 |                                        |
|          383 |                                 |                                        |
|          384 |                                 |                                        |
|          385 |                                 |                                        |
|          386 |                                 |                                        |
|          387 |                                 |                                        |
|         2492 |                                 |                                        |
|         2493 |                                 |                                        |
|         2494 |                                 |                                        |
|         2495 |                                 |                                        |
|         2500 |                                 |                                        |
|         2501 |                                 |                                        |
|         2532 |                                 |                                        |
|         2533 |                                 |                                        |
|         2544 |                                 |                                        |
|         2545 |                                 |                                        |
|         2588 |                                 |                                        |
|         2589 |                                 |                                        |
|         2590 |                                 |                                        |
|         2591 |                                 |                                        |
|         2606 |                                 |                                        |
|         2607 |                                 |                                        |
|         2608 |                                 |                                        |
|         2609 |                                 |                                        |
|         2616 |                                 |                                        |
|         2617 |                                 |                                        |
|         2618 |                                 |                                        |
|         2619 |                                 |                                        |
|         2622 |                                 |                                        |
|         2623 |                                 |                                        |
|         2634 |                                 |                                        |
|         2635 |                                 |                                        |
|         2636 |                                 |                                        |
|         2637 |                                 |                                        |
|         2656 |                                 |                                        |
|         2657 |                                 |                                        |
|         2662 |                                 |                                        |
|         2663 |                                 |                                        |
|         2676 |                                 |                                        |
|         2677 |                                 |                                        |
|         2684 |                                 |                                        |
|         2685 |                                 |                                        |
|         2712 |                                 |                                        |
|         2713 |                                 |                                        |
|         2714 |                                 |                                        |
|         2715 |                                 |                                        |
|         2716 |                                 |                                        |
|         2717 |                                 |                                        |
|         2718 |                                 |                                        |
|         2719 |                                 |                                        |
|         2730 |                                 |                                        |
|         2731 |                                 |                                        |
|         2732 |                                 |                                        |
|         2733 |                                 |                                        |
|         2734 |                                 |                                        |
|         2735 |                                 |                                        |
|         2736 |                                 |                                        |
|         2737 |                                 |                                        |
|         2738 |                                 |                                        |
|         2739 |                                 |                                        |
|         2740 |                                 |                                        |
|         2741 |                                 |                                        |
|         2742 |                                 |                                        |
|         2743 |                                 |                                        |
|         2746 |                                 |                                        |
|         2747 |                                 |                                        |
|         2748 |                                 |                                        |
|         2749 |                                 |                                        |
|         2916 |                                 |                                        |
|         2917 |                                 |                                        |
|         2918 |                                 |                                        |
|         2919 |                                 |                                        |
|         2920 |                                 |                                        |
|         2921 |                                 |                                        |
|         2922 |                                 |                                        |
|         2923 |                                 |                                        |
|         2949 |                                 |                                        |
|         2950 |                                 |                                        |
|         2951 |                                 |                                        |
|         2952 |                                 |                                        |
|         2953 |                                 |                                        |
|         2954 |                                 |                                        |
|         2955 |                                 |                                        |
|         2956 |                                 |                                        |
|         2957 |                                 |                                        |
|         2958 |                                 |                                        |
|         3001 |                                 |                                        |
|         3002 |                                 |                                        |
|         3007 |                                 |                                        |
|         3008 |                                 |                                        |
|         3009 |                                 |                                        |
|         3010 |                                 |                                        |
|         3011 |                                 |                                        |
|         3012 |                                 |                                        |
|         3013 |                                 |                                        |
|         3014 |                                 |                                        |
|         3015 |                                 |                                        |
|         3016 |                                 |                                        |
|         3024 |                                 |                                        |
|         3025 |                                 |                                        |
|         3026 |                                 |                                        |
|         3027 |                                 |                                        |
|         3028 |                                 |                                        |
|         3029 |                                 |                                        |
|         3034 |                                 |                                        |
|         3035 |                                 |                                        |
|         3038 |                                 |                                        |
|         3039 |                                 |                                        |
|         3040 |                                 |                                        |
|         3041 |                                 |                                        |
|         3042 |                                 |                                        |
|         3043 |                                 |                                        |
|         3044 |                                 |                                        |
|         3045 |                                 |                                        |
|           66 |                                 | Total Inbound                          |
|           66 |                                 | Current:                               |
|           66 |                                 | Average:                               |
|           66 |                                 | Maximum:                               |
|           67 |                                 | Total Outbound                         |
|           67 |                                 | Current:                               |
|           67 |                                 | Average:                               |
|           67 |                                 | Maximum:                               |
|            0 |                                 |                                        |
|           66 | |95:bits:0:aggregate_current:2| | 95th Percentile In  ->                 |
|           66 |                                 | (|95:bits:6:aggregate_current:2| mbit) |
|           66 | |99:bits:0:aggregate_current:2| | 99th Percentile In  ->                 |
|           66 |                                 | (|99:bits:6:aggregate_current:2| mbit) |
|           66 | |95:bits:0:aggregate_current:2| | 95th Percentile Out ->                 |
|           67 |                                 | (|95:bits:6:aggregate_current:2| mbit) |
|           67 | |99:bits:0:aggregate_current:2| | 99th Percentile Out ->                 |
|           67 |                                 | (|99:bits:6:aggregate_current:2| mbit) |
+--------------+---------------------------------+----------------------------------------+
167 rows in set (0.00 sec)
```

**Comment by DFlexy** [comment]
This is the other one that I removed from the unit to work

```
MariaDB [cactidb]> select task_item_id, value, text_format from graph_templates_item where local_graph_id=1483;
+--------------+---------------------------------+----------------------------------------+
| task_item_id | value                           | text_format                            |
+--------------+---------------------------------+----------------------------------------+
|         3007 |                                 |                                        |
|         3008 |                                 |                                        |
|         3009 |                                 |                                        |
|         3010 |                                 |                                        |
|         3011 |                                 |                                        |
|         3012 |                                 |                                        |
|         3013 |                                 |                                        |
|         3014 |                                 |                                        |
|         3015 |                                 |                                        |
|         3016 |                                 |                                        |
|           66 |                                 |                                        |
|           66 |                                 | Current:                               |
|           66 |                                 | Average:                               |
|           66 |                                 | Maximum:                               |
|           67 |                                 |                                        |
|           67 |                                 | Current:                               |
|           67 |                                 | Average:                               |
|           67 |                                 | Maximum:                               |
|            0 |                                 |                                        |
|           66 | |95:bits:0:aggregate_current:2| | 95th Percentile In  ->                 |
|           66 |                                 | (|95:bits:6:aggregate_current:2| mbit) |
|           66 | |99:bits:0:aggregate_current:2| | 99th Percentile In  ->                 |
|           66 |                                 | (|99:bits:6:aggregate_current:2| mbit) |
|           66 | |95:bits:0:aggregate_current:2| | 95th Percentile Out ->                 |
|           67 |                                 | (|95:bits:6:aggregate_current:2| mbit) |
|           67 | |99:bits:0:aggregate_current:2| | 99th Percentile Out ->                 |
|           67 |                                 | (|99:bits:6:aggregate_current:2| mbit) |
+--------------+---------------------------------+----------------------------------------+
27 rows in set (0.00 sec)

```

**Comment by DFlexy** [comment]
This is from a new chart that is still working without updating the template

```
MariaDB [cactidb]> select task_item_id, value, text_format from graph_templates_item where local_graph_id=1484;
+--------------+---------------------------------+----------------------------------------+
| task_item_id | value                           | text_format                            |
+--------------+---------------------------------+----------------------------------------+
|         2959 |                                 |                                        |
|         2960 |                                 |                                        |
|         2967 |                                 |                                        |
|         2968 |                                 |                                        |
|         2963 |                                 |                                        |
|         2964 |                                 |                                        |
|         2969 |                                 |                                        |
|         2970 |                                 |                                        |
|         3007 |                                 |                                        |
|         3008 |                                 |                                        |
|           66 |                                 |                                        |
|           66 |                                 | Current:                               |
|           66 |                                 | Average:                               |
|           66 |                                 | Maximum:                               |
|           67 |                                 |                                        |
|           67 |                                 | Current:                               |
|           67 |                                 | Average:                               |
|           67 |                                 | Maximum:                               |
|            0 |                                 |                                        |
|           66 | |95:bits:0:aggregate_current:2| | 95th Percentile In  ->                 |
|           66 |                                 | (|95:bits:6:aggregate_current:2| mbit) |
|           66 | |99:bits:0:aggregate_current:2| | 99th Percentile In  ->                 |
|           66 |                                 | (|99:bits:6:aggregate_current:2| mbit) |
|           66 | |95:bits:0:aggregate_current:2| | 95th Percentile Out ->                 |
|           67 |                                 | (|95:bits:6:aggregate_current:2| mbit) |
|           67 | |99:bits:0:aggregate_current:2| | 99th Percentile Out ->                 |
|           67 |                                 | (|99:bits:6:aggregate_current:2| mbit) |
+--------------+-------------
```

**Comment by TheWitness** [comment]
Thanks for that.  There is an error in your template.  Look at the screen shot below.  66 is in, 67 is out.

![image](https://github.com/Cacti/cacti/assets/1439914/190f94bd-2029-43d4-a230-0d61cef75189)

**Comment by TheWitness** [comment]
It's your Graph Template.

**Comment by DFlexy** [comment]
can help check here ?

![image](https://github.com/Cacti/cacti/assets/50220928/4717cf24-f3f4-41ae-9453-08bd243a72ac)

**Comment by TheWitness** [comment]
You must first:

1) Edit the original Graph Template and fix the entry
2) Then go to the Aggregate Template and resave it.

Still waiting on the dump.

**Comment by DFlexy** [comment]
Good morning.
I checked all chart template settings
I removed the aggregate template again and created it again.

Here is what I identified.
I created 3 new aggregates.

1 with 4 interfaces
It works but after saving the aggregate template it stops working

1 with 10 interfaces
It works but after saving the aggregate template it stops working

1 with 20 interfaces
Works smoothly even after saving the aggregated template

**Comment by DFlexy** [comment]
AWAYS WORK
|           66 |                                 |                                        |
|           66 |                                 | Current:                               |
|           66 |                                 | Average:                               |
|           66 |                                 | Maximum:                               |
|           67 |                                 |                                        |
|           67 |                                 | Current:                               |
|           67 |                                 | Average:                               |
|           67 |                                 | Maximum:                               |
|            0 |                                 |                                        |
|           66 | |95:bits:0:aggregate_current:2| | 95th Percentile In  ->                 |
|           66 |                                 | (|95:bits:6:aggregate_current:2| mbit) |
|           66 | |99:bits:0:aggregate_current:2| | 99th Percentile In  ->                 |
|           66 |                                 | (|99:bits:6:aggregate_current:2| mbit) |
|           67 | |95:bits:0:aggregate_current:2| | 95th Percentile Out ->                 |
|           67 |                                 | (|95:bits:6:aggregate_current:2| mbit) |
|           67 | |99:bits:0:aggregate_current:2| | 99th Percentile Out ->                 |
|           67 |                                 | (|99:bits:6:aggregate_current:2| mbit) |
+--------------+---------------------------------+----------------------------------------+

DONT WORK
+--------------+-------+-------------+
| task_item_id | value | text_format |
+--------------+-------+-------------+
|         2408 |       |             |
|         2409 |       |             |
|         2576 |       |             |
|         2577 |       |             |
|         2959 |       |             |
|         2960 |       |             |
|         2969 |       |             |
|         2970 |       |             |
|           66 |       |             |
|           66 |       | Current:    |
|           66 |       | Average:    |
|           66 |       | Maximum:    |
|           67 |       |             |
|           67 |       | Current:    |
|           67 |       | Average:    |
|           67 |       | Maximum:    |
+--------------+-------+-------------+


DONT WORK
+--------------+-------+-------------+
| task_item_id | value | text_format |
+--------------+-------+-------------+
|         2975 |       |             |
|         2976 |       |             |
|         2977 |       |             |
|         2978 |       |             |
|         2979 |       |             |
|         2980 |       |             |
|         2981 |       |             |
|         2982 |       |             |
|         2987 |       |             |
|         2988 |       |             |
|         2989 |       |             |
|         2990 |       |             |
|         2991 |       |             |
|         2992 |       |             |
|         2993 |       |             |
|         2994 |       |             |
|         2995 |       |             |
|         2996 |       |             |
|         3036 |       |             |
|         3037 |       |             |
|           66 |       |             |
|           66 |       | Current:    |
|           66 |       | Average:    |
|           66 |       | Maximum:    |
|           67 |       |             |
|           67 |       | Current:    |
|           67 |       | Average:    |
|           67 |       | Maximum:    |
+--------------+-------+-------------+

**Comment by DFlexy** [comment]
I believe it has something to do with the number of interfaces

**Comment by DFlexy** [comment]
![image](https://github.com/Cacti/cacti/assets/50220928/3567ead1-e519-4b10-aa5d-3fcb8b700ca5)

**Comment by DFlexy** [comment]
New test with 30 interfaces also stopped working, only the one with 20 interfaces that works anyway

**Comment by TheWitness** [comment]
The only way I'll be able to go further on this is to have a db dump.

**Comment by TheWitness** [comment]
Ope! I just reproduced it in my local environment!

**Comment by TheWitness** [comment]
Okay, it's fixed now.  Just re-save the Template once updated.  Thanks for your patience.

**Comment by TheWitness** [comment]
I've logged this enhancement as well.

https://github.com/Cacti/cacti/issues/5595

**Comment by DFlexy** [comment]
Woohoo Thank you very much for your patience and I really appreciate your effort.
Working perfectly.
I played the file and the graphics worked without problems.
Thank you and that I have everything good

![image](https://github.com/Cacti/cacti/assets/50220928/80acb55f-10e8-424e-93ba-359c8ad748f8)
